vault_cli/vault_python_api.py

```diff
@@ -14,15 +15,32 @@
 path = 'secret/'
 
 
+def check_status():
+    """Check Vault service availability"""
+    r = s.head(urljoin(url, 'sys/health'), headers={})
+    if r.status_code != requests.codes.ok:
+        errors = {
+            429: 'unsealed and standby',
+            472: 'data recovery mode replication secondary',
+            501: 'not initialized',
+            503: 'sealed',
+            404: 'wrong URL'
+        }
+        status = errors.get(r.status_code, 'unknown')
+        sys.exit("Vault service isn't available (status: %s)" % status)
+
+
 def userpass_authentication(username, password):
+    check_status()
+
     d = {"password": password}
     r = s.post(url + 'auth/userpass/login/' + username, json=d, headers={})
 
     if r.status_code == requests.codes.ok:
         json_response = r.json()
         s.headers.update({'X-Vault-Token': json_response.get('auth').get('client_token')})
     else:
-        print('Wrong username or password (HTTP code: %s)' % r.status_code)
+        sys.exit('Wrong username or password (HTTP code: %s)' % r.status_code)
```

Comment:
`click.exit()` and you can remove the sys import :)

Reply:
I wasn't able to use `click.exit` could you provide more information how to use it ?